packages/table/src/headers/_headers.scss

```diff
@@ -254,3 +252,32 @@
   }
   // stylelint-enable selector-class-pattern
 }
+
+.bp-table-column-name,
+.bp-table-row-name {
+  transition: color $cell-transition-duration;
+}
+
+.bp-table-header.pt-loading {
+  @include cell-content-align-vertical();
+
+  .bp-table-column-name,
+  .bp-table-row-name {
+    flex: 1;
+  }
+
+  .bp-table-column-name-text {
+    @include cell-content-align-vertical();
+    padding: floor(($column-header-min-height - $column-name-text-skeleton-height) / 2)
+      $pt-grid-size
+      ceil(($column-header-min-height - $column-name-text-skeleton-height) / 2);
```

Comment:
alternately, just set the super simple `padding: $pt-grid-size;` to ensure space on all sides, and let flex box handle the actual centering.